### ISSUE TITLE
u32 is #[reflect_value], but has no InspectorEguiImpl registered in the TypeRegistry.

### ISSUE THREAD

**Issue by GabiJunkes** [jakobhellermann/bevy-inspector-egui#217]
Using `bevy_inspector::ui_for_value` on version `0.26.0` gives:
 `u32 is #[reflect_value], but has no InspectorEguiImpl registered in the TypeRegistry .
Try calling .register_type::<u32> or add the DefaultInspectorConfigPlugin for builtin types.`.

Just changing to `0.25.2` fixes the issue.
My code:

```rust
use bevy::prelude::*;
use bevy_editor_pls::egui;
use bevy_editor_pls_core::editor_window::{EditorWindow, EditorWindowContext};
use bevy_inspector_egui::bevy_inspector;

use crate::world::world_gen::WorldGenerationData;

pub struct MapGenWindow;

impl EditorWindow for MapGenWindow {
    type State = WindowState;
    const NAME: &'static str = "World Generator";

    fn ui(world: &mut World, mut cx: EditorWindowContext, ui: &mut egui::Ui) {
        world.resource_scope(|world, mut world_data: Mut<WorldGenerationData>| {    
            let Some(state) = cx.state_mut::<MapGenWindow>() else {
                return;
            };
    
            ui.label("Generator:");
            ui.indent("Seed", |ui| {
                ui.horizontal(|ui| {
                    ui.label("Seed:");
                    if bevy_inspector::ui_for_value(&mut state.old_seed, ui, world) {
                        world_data.seed = state.old_seed;
                    }
                })
            });
        })
    }
}

pub struct WindowState {
    old_seed: u32,
}

impl Default for WindowState {
    fn default() -> Self {
        WindowState {
            old_seed: 204,
        }
    }
}
```

Version `0.26.0`:
![image](https://github.com/user-attachments/assets/564f51ba-02be-4f5f-9d4b-b70aedef0596)

Version `0.25.2`:
![image](https://github.com/user-attachments/assets/07e80362-6c76-4f6c-b6c6-95401b873dc6)

**Comment by muhuk** [comment]
I'm getting the same issue with `NonZeroU16` and version `0.25.2`.

![image](https://github.com/user-attachments/assets/7b325c30-2b2a-4f22-8920-3a94bf293799)

**Comment by GabiJunkes** [comment]
Adding `.register_type_data::<u32, InspectorEguiImpl>()` fixed the issue with version `0.26.0`.
